### PR TITLE
Print warning when the Eth1 chain is shorter than follow distance

### LINKED ISSUE #1888: Eth1DepositManager error when chain is shorter than min follow distance
Eth1DataManager repeatedly logs errors when the eth1 chain is shorter than the min follow distance.

Run besu with:

`besu --rpc-http-enabled=true --genesis-file=/Users/aj/Documents/code/teku/acceptance-tests/src/testFixtures/resources/besu/depositContractGenesis.json --miner-enabled --miner-coinbase=0xfe3b557e8fb62b89f4916b721be55ceb828dbd73 --rpc-http-port=8546 --data-path=/tmp/besu`

and then Teku with:

`  master ●  ./build/install/teku/bin/teku --network schlesi --eth1-deposit-contract-address=0xdddddddddddddddddddddddddddddddddddddddd --eth1-endpoint=http://localhost:8546/ --log-destination=console --p2p-enabled=false`

and it will repeated print errors like:

`20:02:28.772 WARN  - Eth1DepositManager failed to get the head of Eth1. Retrying in 2 seconds.`

### PR DESCRIPTION
**Pull request by cemozerr**
## PR Description
Print warning when the Eth1 chain is shorter than follow distance

## Fixed Issue(s)

Fixes #1888

## Documentation

- [x] I thought about documentation and added the `documentation` label to this PR if updates are required.

### CODE REVIEW

File: pow/src/main/java/tech/pegasys/teku/pow/Eth1DepositManager.java [hunk 1]
```diff
@@ -146,8 +146,16 @@ private void preGenesisSubscription(final BigInteger lastProcessedBlockNumber) {
     return eth1Provider
         .getLatestEth1Block()
         .thenApply(EthBlock.Block::getNumber)
-        .thenApply(number -> number.subtract(Constants.ETH1_FOLLOW_DISTANCE.bigIntegerValue()))
         .thenApply(UInt64::valueOf)
+        .thenApply(
+            number -> {
+              if (number.isLessThan(Constants.ETH1_FOLLOW_DISTANCE)) {
```
Comment (mbaxter) [inline pow/src/main/java/tech/pegasys/teku/pow/Eth1DepositManager.java]: Should we retry less frequently and log a warning in this case?

Comment (cemozerr) [reply pow/src/main/java/tech/pegasys/teku/pow/Eth1DepositManager.java]: Makes sense, will do that.

Comment (ajsutton) [reply pow/src/main/java/tech/pegasys/teku/pow/Eth1DepositManager.java]: Not sure we even need a warning - it's a pretty expected condition when using a local test chain.  And we could just start following from 0 now I think about it.  The genesis block of ETH1 is finalized so there's no risk of it being re-orged.  We should then just sit at block 0 until the chain reaches 1024 blocks long where we can carry on as normal.

Comment (cemozerr) [reply pow/src/main/java/tech/pegasys/teku/pow/Eth1DepositManager.java]: > We should then just sit at block 0 until the chain reaches 1024 blocks long where we can carry on as normal.

That made sense to me, so I made the changes to reflect that.

**Review by mbaxter — APPROVED** [review]
LGTM

**Pull request merged**